overlays/keystone/board/sifive/hifive-unmatched/patches/uboot/0001-Implement-keystone-secureboot-feature.patch

```diff
@@ -0,0 +1,5217 @@
+From 15fe2ac799dfd094d01faaf6507eb4944cb3dda2 Mon Sep 17 00:00:00 2001
```

Comment:
Woah, this is really cool! Thank you for this change -- exciting to see someone working on making secure boot work well on this platform :)

I wonder if integrating this feature would almost work better if we split apart the new files you're _adding_ from the files you're _modifying_. Roughly, all of the files under `arch/riscv/lib/` (which seem to be from an upstream project?) could be copied/submoduled into the Keystone project, let's say at `overlays/keystone/board/sifive/hifive-unmatched/src/uboot-ed25519` or something. Then, in `overlays/keystone/external.mk`, you can add something like this (not tested, general idea):

```
ifeq ($(KEYSTONE_PLATFORM),unmatched)

define UBOOT_COPY_HIFIVE_SOURCES
        mkdir -p $(@D)/arch/riscv/lib/ed25519
        cp -ar $(KEYSTONE)/overlays/keystone/board/sifive/hifive-unmatched/src/uboot-ed25519/* $(@D)/arch/riscv/lib/ed25519
endef

UBOOT_POST_EXTRACT_HOOKS += UBOOT_COPY_HIFIVE_SOURCES

endif
```

Then, this patch file would only contain your changes to `arch/riscv/Kconfig`, `arch/riscv/lib/Makefile`, and `common/spl/spl_opensbi.c` which seem to focus on integrating `ed25519` into Keystone. This way, we separate out new functionality from new sources in a way that seems more maintainable.

Reply:
That's make sense.
I am not familiar with the buildroot build system, so I could not come up with the best solution.
I'll try to ensure that only changes to existing sources are included in the patch.

Reply:
New files are located in `board/sifive/hifive-unmatched/src/uboot`. They are copied into source when extracted.
Now, the patch includes only modification in original u-boot source.
This works well.

Reply:
Thank you! This looks great